lib/memo_wise.rb

```diff
@@ -34,7 +34,11 @@ def memo_wise(method_name)
 
         module_eval <<-END_OF_METHOD, __FILE__, __LINE__ + 1
           def #{method_name}(*args)
-            key = [:#{method_name}, args].freeze
+            key = if args.empty?
```

Comment:
@jemmaissroff how would you feel about pulling this logic outside the method so it's not getting called every time? I'm thinking something like:

```ruby
if method(:method_name).arity.zero?
  method_name_and_args = method_name
  key = ":#{method_name}"
else
  method_name_and_args = "#{method_name}(*args)"
  key = "[:#{method_name}, args].freeze"
end

module_eval <<-END_OF_METHOD, __FILE__, __LINE__ + 1
  def #{method_name_and_args}
    key = #{key}
    ...
```

though I'd also be open to just having separate `module_eval`s based on the different versions.

Reply:
Also this is the most nitpicky thing but I'm realizing now that `method_name.inspect` may be a hair more efficient than `":#{method_name}"` because it's able to more easily allocate exactly the string size it needs. (Though to be honest I don't have a great sense for everything [the source](https://ruby-doc.org/core-2.7.1/Symbol.html#method-i-inspect) is doing.)

Reply:
This is really neat! And looks like makes us faster than the other gems for the `no_args` case too! Pushed another version here.